packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart

```diff
@@ -325,10 +326,12 @@ final class SignInStateMachine
         createSmsMfaRequest(event),
       ChallengeNameType.softwareTokenMfa when hasUserResponse =>
         createSoftwareTokenMfaRequest(event),
+      ChallengeNameType.emailOtp when hasUserResponse =>
+        createEmailMfaRequest(event),
       ChallengeNameType.selectMfaType when hasUserResponse =>
         createSelectMfaRequest(event),
-      ChallengeNameType.mfaSetup when hasUserResponse =>
-        createMfaSetupRequest(event),
+      ChallengeNameType.mfaSetup =>
+        handleMfaSetup(event: event, hasUserResponse: hasUserResponse),
```

Comment:
```suggestion
      ChallengeNameType.mfaSetup  when hasUserResponse=>
        handleMfaSetup(event: event),
```
so that `handleMfaSetup` does not need to check `hasUserResponse` and return null if `hasUserResponse == false`